### PR TITLE
[BREAKING CHANGES] - REQUIRES TESTING -- Eventargs interfaces + Global code style application

### PR DESCRIPTION
**Pull request by joker-119**
All EventArg classes have been moved into folders corresponding to what type of event they are, in line with how those events are laid out in the Patches folder.

Additionally, event args used by EXILED are now all based on the `IExiledEvent` interface, instead of `System.EventArgs`.
This allows us to create even more Interfaces for commonly used properties, such as ev.Player, ev.Target, ev.DamageHandler, ev.Item, ev.Firearm and so on.
These properties now belong to their appropriate interface, and the EventArgs classes used by events will inherit the interfaces it needs, then add whatever other properties the event needs ontop of those.

For most plugin devs, aside from all events having standardized property naming and new namespaces (yes, this will break just about every plugin using Exiled.Events), they won't even know the difference between the old and new EventArgs system.

However for those of us who occasionally have to go through this hell:

```
 protected override void SubscribeEvents()
    {
        Exiled.Events.Handlers.Player.InteractingElevator += OnInteractingElevator;
        Exiled.Events.Handlers.Player.InteractingDoor += OnInteractingDoor;
        Exiled.Events.Handlers.Player.InteractingLocker += OnInteractingLocker;
        Exiled.Events.Handlers.Player.InteractingScp330 += OnInteractingScp330;
        Exiled.Events.Handlers.Player.InteractingShootingTarget += OnInteractingShootingTarget;
        Exiled.Events.Handlers.Player.PickingUpItem += OnPickingUpItem;
        Exiled.Events.Handlers.Player.EnteringFemurBreaker += OnEnteringFemurBreaker;
        Exiled.Events.Handlers.Player.Shooting += OnShooting;
        Exiled.Events.Handlers.Player.Hurting += OnHurting;
        Exiled.Events.Handlers.Scp096.AddingTarget += OnAddingTarget;
        Exiled.Events.Handlers.Scp106.Containing += OnContaining;
        Exiled.Events.Handlers.Scp914.Activating += OnActivating914;
        Exiled.Events.Handlers.Scp914.UpgradingPlayer += OnUpgradingPlayer;
        Exiled.Events.Handlers.Scp914.UpgradingInventoryItem += OnUpgradingInventoryItem;
        Exiled.Events.Handlers.Player.PickingUpAmmo += OnPickingUpAmmo;
        Exiled.Events.Handlers.Player.PickingUpArmor += OnPickingUpArmor;
        base.SubscribeEvents();
    }
    
        private void OnPickingUpArmor(PickingUpArmorEventArgs ev)
    {
        if (Check(ev.Player))
            ev.IsAllowed = false;
    }

    private void OnPickingUpAmmo(PickingUpAmmoEventArgs ev)
    {
        if (Check(ev.Player))
            ev.IsAllowed = false;
    }

    private void OnUpgradingInventoryItem(UpgradingInventoryItemEventArgs ev)
    {
        if (Check(ev.Player))
            ev.IsAllowed = false;
    }

    private void OnUpgradingPlayer(UpgradingPlayerEventArgs ev)
    {
        if (Check(ev.Player))
            ev.IsAllowed = false;
    }

    private void OnActivating914(ActivatingEventArgs ev)
    {
        if (Check(ev.Player))
            ev.IsAllowed = false;
    }
...
```
To do things like, make someone unable to interact with the game world in any way, instead, we can do something like this:

```
 protected override void SubscribeEvents()
    {
        Exiled.Events.Handlers.Player.InteractingElevator += OnPlayerEvent;
        Exiled.Events.Handlers.Player.InteractingDoor += OnPlayerEvent;
        Exiled.Events.Handlers.Player.InteractingLocker += OnPlayerEvent;
        Exiled.Events.Handlers.Player.InteractingScp330 += OnPlayerEvent;
        Exiled.Events.Handlers.Player.InteractingShootingTarget += OnPlayerEvent;
        Exiled.Events.Handlers.Player.PickingUpItem += OnPlayerEvent;
        Exiled.Events.Handlers.Player.EnteringFemurBreaker += OnPlayerEvent;
        Exiled.Events.Handlers.Player.Shooting += OnPlayerEvent;
        Exiled.Events.Handlers.Player.Hurting += OnPlayerEvent;
        Exiled.Events.Handlers.Scp096.AddingTarget += OnPlayerEvent;
        Exiled.Events.Handlers.Scp106.Containing += OnPlayerEvent;
        Exiled.Events.Handlers.Scp914.Activating += OnPlayerEvent;
        Exiled.Events.Handlers.Scp914.UpgradingPlayer += OnPlayerEvent;
        Exiled.Events.Handlers.Scp914.UpgradingInventoryItem += OnPlayerEvent;
        Exiled.Events.Handlers.Player.PickingUpAmmo += OnPlayerEvent;
        Exiled.Events.Handlers.Player.PickingUpArmor += OnPlayerEvent;
        base.SubscribeEvents();
    }

  private void OnPlayerEvent(IExiledEvent eventArgs)
  {
      if (eventArgs is IPlayerEvent and IDeniableEvent ev)
          if (Check(ev.Player))
              ev.IsAllowed = false;
}
```

This also forces standardization of property naming for events using similar properties.
Devs will know that any event involving a player, will have ev.Player
Any event related to a gun will have ev.Firearm
etc.

I have also (tried my best to, at least) imported the code style preferences from StyleCop into my ide and blanketed the code style over everything. This should result in a much more consistent look and feel to the code that future contributors can follow, rather than everything being a mish-mash of everyones idiosyncrasies with contradicting style types all over the place.

### CODE REVIEW

**Comment by joker-119** [comment]
In my very limited testing, with a custom build of DevTools to switch from EventArgs to IExiledEvent, events still seem to fire and their property values are what's expected, however I have not yet done any throughout testing to make sure changing the values still works, etc.

File: Exiled.Events/EventArgs/Player/DamagingShootingTargetEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        /// </summary>
+        public Vector3 HitLocation { get; }
+
+        /// <summary>
+        ///     Gets or sets the damage amount.
+        /// </summary>
+        public float Amount { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the distance between the shooter and the shooting target.
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Player/DamagingShootingTargetEventArgs.cs:83]: Gets or sets the distance between the player and the shooting target.

**Review by louis1706 — COMMENTED** [review]
I will test the code this night with my plugin :)
( I am pretty sure nothing are break)

File: Exiled.Events/EventArgs/Map/ChangingIntoGrenadeEventArgs.cs [hunk 2]
```diff
@@ -5,51 +5,50 @@
...
+            Type = pickup is not null ? pickup.Info.ItemId : ItemType.None;
             FuseTime = Pickup.Base is TimeGrenade timeGrenade ? timeGrenade._fuseTime : 3f;
         }
 
         /// <summary>
-        /// Gets a value indicating the pickup being changed.
+        ///     Gets or sets a value indicating what type of grenade will be spawned.
         /// </summary>
-        public Pickup Pickup { get; }
+        public ItemType Type { get; set; }
```
[the remark this answers is not in the document]
Comment (joker-119) [reply Exiled.Events/EventArgs/Map/ChangingIntoGrenadeEventArgs.cs:37]: I don't know why a setter got added here, I don't remember doing that lol

[the remark this answers is not in the document]
Comment (louis1706) [reply Exiled.Events/EventArgs/Map/ChangingIntoGrenadeEventArgs.cs:37]: It's was certainly here before you make it

**Review by PintTheDragon — APPROVED** [review]

**Comment by PintTheDragon** [comment]
Tested SCPStats and WaitAndChillReborn (with a modification almost exactly like the example) with @Michal78900, and everything seems to work fine.

**Review by Thundermaker300 — APPROVED** [review]

**Review by NaoUnderscore — APPROVED** [review]

**Comment by Thundermaker300** [comment]
Omg!!!!

**Pull request merged**